### PR TITLE
Redirect when decrypted published

### PR DESCRIPTION
**Pull request by jon-acker**
### What

Handle case were file is already decrypted and user should be redirected to public-decrypted S3 bucket.

### CODE REVIEW

File: api/download_test.go [hunk 1]
```diff
@@ -84,11 +84,25 @@ func TestHandleFileNotPublished(t *testing.T) {
...
+			h := api.CreateV1DownloadHandler(fetchMetadata, downloadFile, "")
 
 			h.ServeHTTP(rec, req)
 
 			assert.Equalf(t, tt.expectedStatus, rec.status, "CreateV1DownloadHandler(%v)", tt.name)
 		})
 	}
 }
+
+func TestHandlingErrorWhenBadPublicBucketUrl(t *testing.T) {
```
Comment (goofballLogic) [inline api/download_test.go:96]: is the intent to test unhappy path at this level and happy path at the component-test level?

File: files/metadata.go [hunk 2]
```diff
@@ -37,3 +37,7 @@ func (m Metadata) GetContentLength() string {
 func (m Metadata) Unpublished() bool {
 	return m.State == UPLOADED || m.State == CREATED
 }
+
+func (m Metadata) Decrypted() bool {
```
Comment (goofballLogic) [inline files/metadata.go:41]: TDD/BDD for these states?

File: api/download_test.go (hunk 1 above)
Comment (jon-acker) [reply api/download_test.go]: yes, but we've moved the error checking to the config now, so it will check on app startup instead

**Review by braddle — APPROVED** [review]
LGTM

**Pull request merged**